src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -749,8 +749,13 @@ private protected Compilation RunGenerators(Compilation input, ParseOptions pars
             }
 
             driver ??= CreateGeneratorDriver(parseOptions, generators, analyzerConfigOptionsProvider, additionalTexts);
-            driver = driver.RunGeneratorsAndUpdateCompilation(input, out var compilationOut, out var diagnostics);
-            generatorDiagnostics.AddRange(diagnostics);
+
+            driver = driver.RunGenerators(input);
+            var results = driver.GetRunResult();
+
+            var treeRoot = this.Arguments.GeneratedFilesOutputDirectory ?? Arguments.OutputDirectory;
```

Comment:
This makes me nervous because toggling `$(EmitCompilerGeneratedFiles)` will end up impacting which directory we use here. That ends up impacting determinism because it's possible `/pathmap` accounts for these directories differently. 

Not sure how do this better though. Only thought I had is should we only give them full paths if `GeneratedFilesOutputDirectory` is set? Basically if the user wants full paths they have to help us out here.

Reply:
Yeah, this is the crux of the issue really, but @tmat was fairly explicit that they should *always* have full paths as that's whats causing downstream tooling breaks.

We do have a test in this PR that validates `/pathmap` is working as expected for `GeneratedFilesOutputDirectory`, so we should be good on that front. If the user turns on `$(EmitCompilerGeneratedFiles)` then yes the user will have to also path map the directory we use there (which is user controllable too)

Reply:
Seems like a logical requirement for `/pathmap` that it needs to cover the directory specified in `GeneratedFilesOutputDirectory`. In practice, it would likely be some subdirectory of the project directory.